tests/functional/replication/queueProcessor.js

```diff
@@ -362,28 +378,50 @@ describe('queue processor error management with mocking', () => {
 
         // don't call start() on the queue processor, so that we don't
         // attempt to fetch entries from kafka
-
-        s3mock = new S3Mock();
-        httpServer = http.createServer(
-            (req, res) => s3mock.onRequest(req, res));
-        httpServer.listen(7777);
     });
 
     afterEach(() => {
         s3mock.resetTest();
+        httpServer.close();
```

Comment:
ok, that works, I think you could also reassign a new request handler for each test to avoid respawning the server, anyway I'm reworking the instrumentation as part of the error management work so that I think it will change a bit how things are setup.